### PR TITLE
Change products.core to lowercase "self-hosted"

### PR DESCRIPTION
**Pull request by rmloveland**
Formerly, this variable was capitalized as "Self-Hosted" which could be confusing since it implied the term was a proper noun. However, it is not. There is no such Product Name (tm), it's just an adjective referring to the deployment model.

Related to DOC-10665

### CODE REVIEW

**Comment by rmloveland** [comment]
@jaiayu tagging you FYI, no need to review this two character change :-) 

@taroface thanks for the docs team discussion that prompted this - @jaiayu and I discussed with Product Marketing in DMs, and Ayushi decided let's just go with lowercase for now and reassess, and PMM was ok with that

**Comment by taroface** [comment]
This is the change we discussed. Also as discussed, though, it may lead to some unintentional or awkward wordings, like [here](https://deploy-preview-19083--cockroachdb-docs.netlify.app/docs/stable/datadog):

```
The CockroachDB self-hosted integration with Datadog enables data collection and alerting...
```

The way this is phrased, it reads more as though the integration (not CockroachDB) is self-hosted. I'm wondering if there is a way to mitigate some of these instances. It appears that there are just a handful of instances of `The CockroachDB {{ site.data.products.core }}` on each docs branch that could be misread this way. "The self-hosted integration", "The self-hosted Production Checklist".

Another thing is that we still have some occurrences of `Self-Hosted` because people (like me) forgot to use the product variable. We should find-replace those occurrences.

**Comment by rmloveland** [comment]
> This is the change we discussed. Also as discussed, though, it may lead to some unintentional or awkward wordings, like [here](https://deploy-preview-19083--cockroachdb-docs.netlify.app/docs/stable/datadog):
> 
> ```
> The CockroachDB self-hosted integration with Datadog enables data collection and alerting...
> ```
> 
> The way this is phrased, it reads more as though the integration (not CockroachDB) is self-hosted. I'm wondering if there is a way to mitigate some of these instances. It appears that there are just a handful of instances of `The CockroachDB {{ site.data.products.core }}` on each docs branch that could be misread this way. "The self-hosted integration", "The self-hosted Production Checklist".

Good point, I have gone through and tried to edit all of these instances so they read a bit more naturally as "The integration of your CRDB self-hosted cluster with Datadog"

> Another thing is that we still have some occurrences of `Self-Hosted` because people (like me) forgot to use the product variable. We should find-replace those occurrences.

I have find-replaced them in all cases except where they occur in the YAML front matter of the page, in which case I have edited them to say "self-hosted" (lowercase) so they do not appear to be Product Names (tm)

NB. i have applied all of these changes across v23.1, v23.2, v24.1, v24.2, v24.3

PTAL!

File: src/current/v24.1/authentication.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Authenticate to CockroachDB Self-Hosted Clusters
+title: Authenticate to CockroachDB self-hosted Clusters
```
Comment (taroface) [inline src/current/v24.1/authentication.md:2]:
```suggestion
title: Authenticate to CockroachDB Self-Hosted Clusters
```

File: src/current/v24.1/essential-alerts-self-hosted.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
-title: Essential Alerts for CockroachDB Self-Hosted Deployments
-summary: Learn about the recommended essential alerts for monitoring your CockroachDB Self-Hosted cluster.
+title: Essential Alerts for CockroachDB self-hosted Deployments
```
Comment (taroface) [inline src/current/v24.1/essential-alerts-self-hosted.md:2]:
```suggestion
title: Essential Alerts for CockroachDB Self-Hosted Deployments

```

File: src/current/v24.1/essential-metrics-self-hosted.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Essential Metrics for CockroachDB Self-Hosted Deployments
-summary: Learn about the recommended essential metrics for monitoring your CockroachDB Self-Hosted cluster.
+title: Essential Metrics for CockroachDB self-hosted Deployments
```
Comment (taroface) [inline src/current/v24.1/essential-metrics-self-hosted.md:2]:
```suggestion
title: Essential Metrics for CockroachDB Self-Hosted Deployments
```

File: src/current/v24.2/authentication.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Authenticate to CockroachDB Self-Hosted Clusters
+title: Authenticate to CockroachDB self-hosted Clusters
```
Comment (taroface) [inline src/current/v24.2/authentication.md:2]:
```suggestion
title: Authenticate to CockroachDB Self-Hosted Clusters
```

File: src/current/v24.2/auto-tls.md [hunk 5]
```diff
@@ -139,7 +139,7 @@ cockroach init --certs-dir=certs --host=localhost:26257
 
 - [`cockroach cert`]({% link {{ page.version.version }}/cockroach-cert.md %})
 - [`cockroach auth-session`]({% link {{ page.version.version }}/cockroach-auth-session.md %})
-- [Authenticate to CockroachDB Self-Hosted Clusters]({% link {{ page.version.version }}/authentication.md %})
+- [Authenticate to CockroachDB {{ site.data.products.core }} Clusters]({% link {{ page.version.version }}/authentication.md %})
```
Comment (taroface) [inline src/current/v24.2/auto-tls.md:142]: This makes the linked text inconsistent with the title-case page title, but I'm OK with it. Perhaps lowercase the `Clusters` so it looks less weird?

File: src/current/v24.2/certificate-based-authentication-using-the-x509-subject-field.md [hunk 6]
```diff
@@ -118,7 +118,7 @@ Regardless of this setting's value, CockroachDB will verify the following during
 
 - [`ALTER ROLE ... SUBJECT`]({% link {{ page.version.version }}/alter-role.md %}#set-the-subject-role-option-for-certificate-based-authentication)
 - [`CREATE ROLE ... SUBJECT`]({% link {{ page.version.version }}/create-role.md %}#set-the-subject-role-option-for-certificate-based-authentication)
-- [Authenticate to CockroachDB Self-Hosted Clusters]({% link {{ page.version.version }}/authentication.md %})
+- [Authenticate to CockroachDB {{ site.data.products.core }} Clusters]({% link {{ page.version.version }}/authentication.md %})
```
Comment (taroface) [inline src/current/v24.2/certificate-based-authentication-using-the-x509-subject-field.md:121]: Ditto

File: src/current/v24.2/datadog.md [hunk 7]
```diff
@@ -1,17 +1,17 @@
 ---
-title: Monitor CockroachDB Self-Hosted with Datadog
+title: Monitor CockroachDB self-hosted clusters with Datadog
```
Comment (taroface) [inline src/current/v24.2/datadog.md:2]:
```suggestion
title: Monitor CockroachDB Self-Hosted Clusters with Datadog
```

File: src/current/v24.2/encryption.md [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Manage Encryption for CockroachDB Self-Hosted
+title: Manage Encryption for CockroachDB self-hosted clusters
```
Comment (taroface) [inline src/current/v24.2/encryption.md:2]:
```suggestion
title: Manage Encryption for CockroachDB Self-Hosted Clusters
```

File: src/current/v24.2/essential-alerts-self-hosted.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
 ---
-title: Essential Alerts for CockroachDB Self-Hosted Deployments
-summary: Learn about the recommended essential alerts for monitoring your CockroachDB Self-Hosted cluster.
+title: Essential Alerts for CockroachDB self-hosted Deployments
```
Comment (taroface) [inline src/current/v24.2/essential-alerts-self-hosted.md:2]:
```suggestion
title: Essential Alerts for CockroachDB Self-Hosted Deployments

```

File: src/current/v24.2/essential-metrics-self-hosted.md [hunk 10]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Essential Metrics for CockroachDB Self-Hosted Deployments
-summary: Learn about the recommended essential metrics for monitoring your CockroachDB Self-Hosted cluster.
+title: Essential Metrics for CockroachDB self-hosted Deployments
```
Comment (taroface) [inline src/current/v24.2/essential-metrics-self-hosted.md:2]:
```suggestion
title: Essential Metrics for CockroachDB Self-Hosted Deployments
```

File: src/current/v24.2/sso-sql.md [hunk 11]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Cluster Single Sign-on (SSO) using JSON web tokens (JWTs)
```
Comment (taroface) [inline src/current/v24.2/sso-sql.md:2]:
```suggestion
title: Cluster Single Sign-on (SSO) using JSON Web Tokens (JWTs)
```

File: src/current/v24.3/authentication.md [hunk 12]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Authenticate to CockroachDB Self-Hosted Clusters
+title: Authenticate to CockroachDB self-hosted Clusters
```
Comment (taroface) [inline src/current/v24.3/authentication.md:2]:
```suggestion
title: Authenticate to CockroachDB Self-Hosted Clusters
```

File: src/current/v24.3/datadog.md [hunk 13]
```diff
@@ -1,17 +1,17 @@
 ---
-title: Monitor CockroachDB Self-Hosted with Datadog
+title: Monitor CockroachDB self-hosted clusters with Datadog
```
Comment (taroface) [inline src/current/v24.3/datadog.md:2]:
```suggestion
title: Monitor CockroachDB Self-Hosted Clusters with Datadog
```

File: src/current/v24.3/encryption.md [hunk 14]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Manage Encryption for CockroachDB Self-Hosted
+title: Manage Encryption for CockroachDB self-hosted clusters
```
Comment (taroface) [inline src/current/v24.3/encryption.md:2]:
```suggestion
title: Manage Encryption for CockroachDB Self-Hosted Clusters
```

File: src/current/v24.3/essential-alerts-self-hosted.md [hunk 15]
```diff
@@ -1,7 +1,7 @@
 ---
-title: Essential Alerts for CockroachDB Self-Hosted Deployments
-summary: Learn about the recommended essential alerts for monitoring your CockroachDB Self-Hosted cluster.
+title: Essential Alerts for CockroachDB self-hosted Deployments
```
Comment (taroface) [inline src/current/v24.3/essential-alerts-self-hosted.md:2]:
```suggestion
title: Essential Alerts for CockroachDB Self-Hosted Deployments

```

File: src/current/v24.3/essential-metrics-self-hosted.md [hunk 16]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Essential Metrics for CockroachDB Self-Hosted Deployments
-summary: Learn about the recommended essential metrics for monitoring your CockroachDB Self-Hosted cluster.
+title: Essential Metrics for CockroachDB self-hosted Deployments
```
Comment (taroface) [inline src/current/v24.3/essential-metrics-self-hosted.md:2]:
```suggestion
title: Essential Metrics for CockroachDB Self-Hosted Deployments
```

File: src/current/v24.3/sso-sql.md [hunk 17]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Cluster Single Sign-on (SSO) using JSON web tokens (JWTs)
```
Comment (taroface) [inline src/current/v24.3/sso-sql.md:2]:
```suggestion
title: Cluster Single Sign-on (SSO) using JSON Web Tokens (JWTs)
```

File: src/current/v24.2/auto-tls.md (hunk 5 above)
Comment (rmloveland) [reply src/current/v24.2/auto-tls.md]: Done!

File: src/current/v24.2/certificate-based-authentication-using-the-x509-subject-field.md (hunk 6 above)
Comment (rmloveland) [reply src/current/v24.2/certificate-based-authentication-using-the-x509-subject-field.md]: Done!

**Comment by rmloveland** [comment]
@taroface applied all suggested changes, PTAL!

**Review by taroface — APPROVED** [review]
LGTM!!

**Pull request merged**